functions/Get-DbaServerRole.ps1

```diff
@@ -0,0 +1,139 @@
+function Get-DbaServerRole {
+	<#
+		.SYNOPSIS
+			Gets the list of server-level roles with the logins that are members of that role.
+
+		.DESCRIPTION
+			Gets the list of server-level role for SQL Server instance. Output will include the logins that are members of the server-level role(s).
+
+		.PARAMETER SqlInstance
+			The SQL Server instance. Server version must be SQL Server version 2005 or higher.
+
+		.PARAMETER SqlCredential
+			Allows you to login to servers using SQL Logins instead of Windows Authentication (AKA Integrated or Trusted). To use:
+
+			$scred = Get-Credential, then pass $scred object to the -SqlCredential parameter.
+
+			Windows Authentication will be used if SqlCredential is not specified. SQL Server does not accept Windows credentials being passed as credentials.
+
+			To connect as a different Windows user, run PowerShell as that user.
+
+		.PARAMETER ServerRole
+			Server-Level role to filter results to that role only.
+
+		.PARAMETER ExcludeServerRole
+			Server-Level role to exclude from results.
+
+		.PARAMETER ExcludeFixedRole
+			Filter the fixed server-level roles. Only applies to SQL Server 2017 that supports custom server-level roles.
+
+		.PARAMETER Login
+			SQL Server login to filter results, will only return roles where the login(s) are a member.
+
+		.PARAMETER EnableException
+			By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message. This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting. Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+		.NOTES
+			Tags: ServerRole, Security
+			Original Author: Shawn Melton (@wsmelton)
+
+			Website: https: //dbatools.io
+			Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+			License: GNU GPL v3 https: //opensource.org/licenses/GPL-3.0
+
+		.LINK
+			https://dbatools.io/Get-DbaServerRole
+
+		.EXAMPLE
+			Get-DbaServerRole -SqlInstance sql2016a
+
+			Outputs list of server-level roles and logins that are members for sql2016a instance.
+
+		.EXAMPLE
+			Get-DbaServerRole -SqlInstance sql2016a -Role sysadmin
+
+			Outputs members of sysadmin server-level role on sql2016a instance.
+
+		.EXAMPLE
+			Get-DbaServerRole -SqlInstance sql2017a -ExcludeFixedRole
+
+			Outputs the user server-level role(s) on sql2017a instance. A SQL Server 2017 supported feature only.
```

Comment:
I'm no sure 'user  server-level role' is the right expression. IIRC 'fixed roles' is not the same as 'system objects'.

Reply:
2017 allows you to create custom roles. You only know that they are custom roles based on the IsFixedRole being false.

Reply:
yes, but [public] is not a fixed role, yet I wouldn't call that a 'user role'

Reply:
Public is a fixed role.

Reply:
https://docs.microsoft.com/en-us/sql/relational-databases/security/authentication-access/server-level-roles#fixed-server-level-roles

Reply:
strange:

![image](https://user-images.githubusercontent.com/22813386/33014301-2e6614a4-cde7-11e7-9feb-d9f2610c32b3.png)